### ISSUE TITLE
Make user router flexible enough to pick few routes only

### ISSUE THREAD

**Issue by jd-solanki** [fastapi-users/fastapi-users#1466]
Hi 👋🏻 

I was using fastapi users in one of my project and tried my best to customize according to requirement however I now come to deadend where `fastapi_users.get_users_router` includes multiple routes and we can't pick from one of the route or exlucde some of them.

In my case, I would like to change dependency of user patch route `PATCH /users/{id}` to change from superuser to authenticated user but I can't change that and also can't exclude that specific route while including router.

**Comment by hgalytoby** [comment]
Copy fastapi_users/router/users.py

Modify https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/router/users.py#L135

**Comment by jd-solanki** [comment]
Already copied but this is not preferable solution. This is kind of patch.

We just loose the updates of the lib.

**Comment by hgalytoby** [comment]
Indeed, but unfortunately, we can only opt for similar approaches to solve the problem.

**Comment by Abdeldjalil-H** [comment]
@hgalytoby @jd-solanki 

I have an idea, I don't know if it will work or it has some drawbacks. If you define the methods like `me`, `update_me`,... outisde the function `get_users_router`, one can register those functions and maybe add more functionalities to it. For example

```python
from fastapi_users.router.users import update_me

@router.patch('/users/{id}') # add dependencies here
async def update_current_user(...):
    await update_me()
    # some other logic
```

You may also extract the `responses` to some variables, so they can be easily reused.

**Comment by arwindersngh62** [comment]
I think this may be out of scope for user management but would fall within the responsibility of an access control or authorization system. If that is the case you van use some sort of policy based system which runs after the route is called. From what you asked I think you are trying to patch in the current user to a different user/type by patching a diff user_id but that should be done **within** the route function using  a dependency that returns a new userId. The dependency can take the superuserID and then return the corresponding ID for an authenticated user. Please let me know if i have misunderstood the problem.

**Comment by cglacet** [comment]
I agree that the current solution is a bit weird as it mixes admin routes with regular user routes. It would make much more sense to have the two separated (or have the option to only have admin/regular routes). 

What I mean by admin is: `/{user_id}` routes, the regular user routes being the `/me` routes.

Here is my current solution to split the two:

```python
fastapi_users_routes = cast(
    List[Route], fastapi_users.get_users_router(UserRead, UserUpdate).routes
)
fastapi_non_admin_router = fastapi_users.get_users_router(UserRead, UserUpdate)
fastapi_non_admin_router.routes = cast(
    List[BaseRoute], [r for r in fastapi_users_routes if r.path == "/me"]
)
app.include_router(fastapi_non_admin_router, prefix="/users")

fastapi_admin_router = fastapi_users.get_users_router(UserRead, UserUpdate)
fastapi_admin_router.routes = cast(
    List[BaseRoute], [r for r in fastapi_users_routes if r.path != "/me"]
)
app.include_router(fastapi_admin_router, prefix="/admin/users")
```

**Comment by arwindersngh62** [comment]
Can anyone help me with this because it looks very wrong to me do it like this. Mixing routes and authentication is a bit dangerous. All users should have access to all routes and then based on the authentication info provided, they may get the data or they may get a auth failed or unauthorized access message. If you anyways want to modify routers and routes that is an implementation detail and should be out of scope for user management.  Also, its not the best idea to provide user id in the URL , that is half the info needed to verify the user and can be picked up by data collection procedures by other tabs. 

Maybe I am misunderstanding this but I think only one auth_router is needed and some sort of auth middleware and thats it. The way it is done is forcing me to change my implementation to work with different sets of routes. The part with who gets what route comes under access control and people use different types of access control. Also access controls are largely defined by organizations and not developers. I cant understand how one would go about defining permissions and roles.

**Comment by frankie567** [comment]
Thank you for your interest!

Unfortunately, we're now in maintenance mode, which means we won't add new features and focus only on security and maintenance updates. [[Read more](https://github.com/fastapi-users/fastapi-users/discussions/1543)]
